simulation/src/main/kotlin/it/unibo/collektive/examples/exercises/DistanceToSource.kt

```diff
@@ -0,0 +1,36 @@
+package it.unibo.collektive.examples.exercises
+
+import it.unibo.collektive.aggregate.api.Aggregate
+import it.unibo.collektive.aggregate.api.Aggregate.Companion.neighboring
+import it.unibo.collektive.alchemist.device.sensors.EnvironmentVariables
+import it.unibo.collektive.examples.exercises.searchSource
+import it.unibo.collektive.stdlib.spreading.distanceTo
+
+/**
+ * 2) Compute the distances between any node and the "source" using the adaptive bellman-ford algorithm.
+*/
+
+fun Aggregate<Int>.distanceToSource(environment: EnvironmentVariables): Int {
+    // Individuate source from the previous exercise 
+    searchSource(environment)
+
+    // The `distanceTo` function calculates the minimum distance between adjacent neighboring nodes,
+    // starting from the source node and propagating distances to all other nodes.
+    // If a node is unreachable, its distance is set to `Int.MAX_VALUE`.
+    //
+    // The aggregation operation uses a reduction function that sums the distances of neighboring nodes,
+    // with a limit of `Int.MAX_VALUE` to prevent overflow.
```

Comment:
Avoid this comment if it is already set into the KDoc of `distanceTo`.
If not, please add the KDoc to `distanceTo`.